third_party/nvfuser/test/test_gpu3.cpp

```diff
@@ -6642,6 +6642,70 @@ TEST_F(NVFuserTest, FusionIssue2077_CUDA) {
 
 #endif
 
+TEST_F(NVFuserTest, FusionIssue2372_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tx = TensorViewBuilder()
+                .ndims(5)
+                .contiguity(std::vector<bool>(5, true))
+                .dtype(DataType::Float)
+                .build();
+  fusion.addInput(tx);
+  auto tmean = TensorViewBuilder()
+                   .ndims(1)
+                   .contiguity({true}) // Contiguous input of even size
+                   .dtype(DataType::Float)
+                   .build();
+  fusion.addInput(tmean);
+  auto tvar = TensorViewBuilder()
+                  .ndims(1)
+                  .contiguity({true})
+                  .dtype(DataType::Float)
+                  .build();
+  fusion.addInput(tvar);
+  auto seps = IrBuilder::newScalar(DataType::Double);
+  fusion.addInput(seps);
+
+  auto tmean_bcast = broadcast(tmean, {true, true, true, true, false});
+  auto tmean_expand = expand_as(tmean_bcast, tx);
+  auto diff = sub(tx, tmean_expand);
+  auto regvar = add(tvar, seps);
+  auto invstd = rsqrt(regvar);
+  auto invstd_bcast = broadcast(invstd, {true, true, true, true, false});
+  auto invstd_expand = expand_as(invstd_bcast, tx);
+  auto x_normed = mul(diff, invstd_expand);
+
+  fusion.addOutput(x_normed);
+  // This output is not necessary for a normalization function, but should not
+  // cause compilation to fail
+  fusion.addOutput(tmean); // Contiguous even-size input added as output
+  fusion.addOutput(invstd);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+
+  at::manual_seed(0);
+
+  int C = 2;
+  at::Tensor x = at::randn({1, 5, 5, 5, C}, options);
+  at::Tensor mean = at::randn({C}, options);
+  at::Tensor var = at::rand({C}, options);
+  double eps = 1e-5;
+
+  std::vector<at::IValue> inputs = {x, mean, var, eps};
+
+  auto lparams = schedulePointwise(&fusion, inputs);
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, inputs, lparams);
+  auto cg_outputs = fe.runFusion(inputs, lparams);
+
+  auto eager_diff = x - mean.view({1, 1, 1, 1, -1});
+  auto eager_invstd = at::rsqrt(var + eps).view({1, 1, 1, 1, -1});
+
+  ASSERT_TRUE(at::allclose(cg_outputs[0], eager_diff * eager_invstd));
```

Comment:
Can you use `testValidate`?

Reply:
Good call. I tried converting to use testValidate instead of only checking that the first output matches. This means checking `mean` against `cg_outputs[1]` as well, which is actually failing since it seems the `Fusion` is outputting an empty 0-dimensional second output (`tmean`). Somehow `addOutput(tmean)` is giving us a tensor with `numel()==0` instead of the. I'm investigating this now with a simpler test:
```cpp
TEST_F(NVFuserTest, FusionIssue2372InputOutput_CUDA) {  Fusion fusion;             
  FusionGuard fg(&fusion);                                                     
                                                                            
  auto x = makeContigTensor(1, DataType::Float);
  fusion.addInput(x);
  
  fusion.addOutput(x);
  
  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
                                                   
  at::manual_seed(0);                                                                                                                                                         
  int C = 2;                                                  
  at::Tensor x_at = at::randn({C}, options);              
                                                          
  std::vector<at::IValue> inputs = {x_at};   
                                                              
  auto lparams = schedulePointwise(&fusion, inputs);
  /*
  C++ exception with description "largest_out != nullptr INTERNAL ASSERT FAILED at "../third_party/nvfuser/csrc/scheduler/pointwise.cpp":101, please report a bug to PyTorch.
Exception raised from getPointwiseHeuristics at ../third_party/nvfuser/csrc/scheduler/pointwise.cpp:101 (most recent call first):
  */
                                                                             
  FusionExecutor fe;
  fe.compileFusion(&fusion, inputs, lparams);
  auto cg_outputs = fe.runFusion(inputs, lparams);

  ASSERT_TRUE(at::allclose(cg_outputs[0], x_at));
}
```
This is a different error since the code in the current test does schedule, but it seems related that the scheduler doesn't see any owned outputs (`HeuristicSummaryEntry::data_ptr_` is `nullptr`). I'm currently comparing this to how aliasing works to try and get an idea how to fix this.

Reply:
The schedulers typically use one tensor as a reference of transformations, and it looks like the pointwise scheduler fails to find any reference tensor, likely because of the structure of the fusion.

Reply:
`testValidate` actually skips aliased outputs. So I kept the explicit assert but also added `testValidate` for the other outputs.